### PR TITLE
update project dependencies

### PR DESCRIPTION
**Pull request by mustartt**

### CODE REVIEW

**Comment by mustartt** [comment]
- part of changes for `SEM-457: upgrade to react 16 : dffacae` was cherry picked from the branch `react-16-update` as the `react-16-pr-branch` did not build on my end. 
 - karma (transform with Istanbul) no longer work

File: package.json [hunk 1]
```diff
@@ -74,19 +74,18 @@
...
     "karma-threshold-reporter": "0.1.15",
     "natives": "^1.1.6",
-    "npm-install-peers": "^1.2.1",
-    "numeral": "1.5.3",
-    "pre-commit": "1.1.2",
-    "react-test-kit": "git+https://github.com/smaato/react-test-kit.git#v0.2.0",
+    "npm-install-peers": "^1.2.2",
+    "numeral": "2.0.6",
+    "pre-commit": "1.2.2",
+    "react-test-kit": "git+https://github.com/smaato/react-test-kit.git#27f4ee1c4963e824776d844209f2d51d92c3f188",
```
Comment (Vsury-ui) [inline package.json]: Why version is not specified here ??

Comment (Vsury-ui) [reply package.json]: I mean for this "react-test-kit": "git+https://github.com/smaato/react-test-kit.git#27f4ee1c4963e824776d844209f2d51d92c3f188",

File: src/framework/filterControl/forms/FilterForm.spec.jsx [hunk 2]
```diff
@@ -85,5 +87,44 @@ describe('FilterForm', () => {
...
+          comparisonParameters: {
+            options: [],
+          },
+        }),
+        onAddFilter: () => undefined,
+      };
+
+      const testCase = TestCaseFactory.create(FilterForm, props);
+
+      expect(testCase.findComponents(MixedTypeValueFilterForm));
```
Comment (Vsury-ui) [inline src/framework/filterControl/forms/FilterForm.spec.jsx]: what is expected ??

File: src/framework/services.js [hunk 3]
```diff
@@ -71,6 +75,6 @@ export {
 
 // Test
 
-export {
-  default as CommonAssertions,
-} from './services/test/CommonAssertions';
+// export {
+//   default as CommonAssertions,
+// } from './services/test/CommonAssertions';
```
Comment (Vsury-ui) [inline src/framework/services.js]: remove commented code

**Review by Vsury-ui — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mustartt) [reply package.json]: didn't see that they had a release for v0.3.0, i'll change that

File: src/guide/views/grid/createRows.js [hunk 4]
```diff
@@ -5,6 +5,22 @@ export default function createRows(indexStart, newRowsCount) {
...
+
+  function getRandomCylinders() {
+    const random = getRandomInteger(1, 16);
+    if (random < 9) {
+      return random;
+    }
+    if (random < 13) {
+      return 'XL engine';
+    }
+    return 'XXL engine';
```
Comment (Enigo) [inline src/guide/views/grid/createRows.js:21]: why was this part changed?

**Comment by Enigo** [comment]
okay, I see you've added some new ui elements:

- src/framework/filterControl/forms/DateRangeFilterForm.jsx
- src/framework/filterControl/forms/MixedTypeValueFilterForm.jsx 
- src/framework/radioButtonItem/RadioButtonItem.jsx
- src/framework/radioButtons/RadioButtons.jsx
- src/framework/services/filter/MixedTypeValueFilter.js


where are they coming from, actually?

**Comment by mustartt** [comment]
The changes for `src/framework/services/filter/MixedTypeValueFilter.js` was picked from [https://github.com/smaato/ui-framework/commit/46b271a8955d007a091697834ade348c09903c5f](https://github.com/smaato/ui-framework/commit/46b271a8955d007a091697834ade348c09903c5f) because after updating to React 16, old implementation for the Grid view was not passing the tests, and it has already been fixed in commit. This change is only used in the GitHub page preview examples.

`src/framework/filterControl/forms/DateRangeFilterForm.jsx` and `src/framework/filterControl/forms/MixedTypeValueFilterForm.jsx` was already exported on develop but the original source was missing, so I added them from [https://github.com/smaato/ui-framework/commit/db5ab18a830d41142baff6440af659f28584bb7e](https://github.com/smaato/ui-framework/commit/db5ab18a830d41142baff6440af659f28584bb7e).

And for `src/framework/radioButtonItem/RadioButtonItem.jsx` and `src/framework/radioButtons/RadioButtons.jsx`, those were left in from [https://github.com/smaato/ui-framework/commit/c97316bac9245d22dcf1b53a9e04ab6262626245](https://github.com/smaato/ui-framework/commit/c97316bac9245d22dcf1b53a9e04ab6262626245), but can be removed without any problems.

File: src/guide/views/grid/createRows.js (hunk 4 above)
Comment (mustartt) [reply src/guide/views/grid/createRows.js:21]: This was pulled along side the fix for Grid Example from the commit for MixedTypeValueFilter [https://github.com/smaato/ui-framework/commit/46b271a8955d007a091697834ade348c09903c5f](https://github.com/smaato/ui-framework/commit/46b271a8955d007a091697834ade348c09903c5f) in the github page preview examples. Both the old implementation and this works with the current Grid Example
